File: [23.12]Design-Pattern/Java/src/creational/abstractFactory/BurgerSetFactoryImpl.java
package creational.abstractFactory;

public class BurgerSetFactoryImpl implements BurgerSetFactory{
    @Override
    public BurgerSet makeSet(String type) {
        BurgerSet burgerSet = null;
        switch (type){
            case "BurgerKing" -> burgerSet = new BurgerSet(new BurgerKingHamburger(), new BurgerKingDrink());
            case "Macdonald" -> burgerSet = new BurgerSet(new MacdonaldHamburger(), new MacdonaldDrink());
            default -> System.out.println("해당 버거 세트가 없음");
        }
        return burgerSet;
    }
}

File: [23.12]Design-Pattern/Kotlin/src/main/kotlin/creational/abstractFactory/BurgerSetFactoryImpl.kt
package creational.abstractFactory

class BurgerSetFactoryImpl: BurgerSetFactory {
    override fun makeBurgerSet(type: String): BurgerSet?{
        var burgerSet: BurgerSet? = null

        when(type){
            "Macdonald" -> burgerSet = BurgerSet(MacdonaldBurger(), MacdonaldDrink())
            "BurgerKing" -> burgerSet = BurgerSet(BurgerKingBurger(), BurgerKingDrink())
        }

        return burgerSet
    }
}